Assets/Scripts/Controllers/HeaderController.cs

```diff
@@ -49,13 +70,58 @@ public class HeaderController : MonoBehaviour {
 					headerTitle.text = Config.CurrentlySelectedText;
 					headerValue.text = nodeController.SelectedNode.Title;
 				}
-			}
-		}
+            } else {
+                connectionController.OnConnectionRangeChanged?.Invoke(0, 0, 0);
+            }
+        }
+
+        private void UpdateNodeHeaderAfterConnectionRangeChange(int start, int end, int count) {
+            TextMesh headerConnectionsRangeText = HeaderObject.transform.GetChild(2).GetComponent<TextMesh>();
+
+			if (nodeController.SelectedNode != null && count > 0) {
+				HeaderObject.transform.GetChild(3).gameObject.SetActive(true);
+				
+				// Primary range size & position
+				float primaryRangeWidth = (float)(end > start ? end - start + 1 : count - start + 1) / count * indicatorWidth;
+				targetPrimaryRangeSize = new Vector2(primaryRangeWidth, indicatorHeight);
+				targetPrimaryRangePosition = new Vector3(-(indicatorWidth - primaryRangeWidth) / 2.0f + ((start-1) * indicatorWidth / count), 0, 0);
+
+				if (end <= start) {
+					// Second range size & position
+					float secondRangeWidth = (float)end / count * 12.0f;
+					targetSecondaryRangeSize = new Vector2(secondRangeWidth, indicatorHeight);
+					targetSecondaryRangePosition = new Vector3(-(indicatorWidth - secondRangeWidth) / 2.0f, 0, 0);
+				} else {
+					targetSecondaryRangeSize = new Vector2(0, indicatorHeight);
+					targetSecondaryRangePosition = new Vector3(-indicatorWidth / 2.0f, 0, 0);
+				}
+
+				// Connections range text
+				if (count <= connectionController.ConnectionDistribution.MaxVisibleNumber) {
+					headerConnectionsRangeText.text = string.Format("{0} <color=white>{1} {2}</color>", Config.CurrentConnectionRangeText, Config.AllConnectionRangeText, count);
+				} else {
+					headerConnectionsRangeText.text = string.Format("{0} <color=white>[{1}-{2}/{3}]</color>", Config.CurrentConnectionRangeText, start, end, count);
+				}
+            } else {
+				HeaderObject.transform.GetChild(3).gameObject.SetActive(false);
+				headerConnectionsRangeText.text = string.Empty;
+            }
+        }
 
 		void Update() {
 			if (networkController.IsClient()) {
 				return;
 			}
+			
+			// Connection indicator update (commented out partially working animation)
+			headerIndicatorPrimaryRangeSprite.size = targetPrimaryRangeSize;
+			HeaderObject.transform.GetChild(3).GetChild(0).localPosition = targetPrimaryRangePosition;
+			headerIndicatorSecondaryRangeSprite.size = targetSecondaryRangeSize;
+			HeaderObject.transform.GetChild(3).GetChild(1).localPosition = targetSecondaryRangePosition;
+			//headerIndicatorPrimaryRangeSprite.size = Vector2.Lerp(headerIndicatorPrimaryRangeSprite.size, targetPrimaryRangeSize, Time.deltaTime * 10);
```

Comment:
If we are not using the animation remove the commented code; it is in git history if we decide we need it later